lighthouse-core/report/html/renderer/element-screenshot-renderer.js

```diff
@@ -0,0 +1,237 @@
+        top: screenshotTopVisisbleEdge,
+      },
+      clip: {
+        left: clipRect.left - screenshotLeftVisibleEdge,
+        top: clipRect.top - screenshotTopVisisbleEdge,
+      },
+    };
+  }
+
+  /**
+   * @param {DOM} dom
+   * @param {string} clipId
+   * @param {{top: number, bottom: number, left: number, right: number}} _
+   */
+  static renderClipPath(dom, clipId, {top, bottom, left, right}) {
+    const clipPathSvg = dom.createElement('div');
+    clipPathSvg.innerHTML = `<svg height="0" width="0">
+        <defs>
+          <clipPath id='${clipId}' clipPathUnits='objectBoundingBox'>
+            <polygon points="0,0  1,0  1,${top} 0,${top}" ></polygon>
+            <polygon points="0,${bottom} 1,${bottom} 1,1 0,1" ></polygon>
+            <polygon points="0,${top} ${left},${top} ${left},${bottom} 0,${bottom}" ></polygon>
+            <polygon points="${right},${top} 1,${top} 1,${bottom} ${right},${bottom}" ></polygon>
+          </clipPath>
+        </defs>
+      </svg>`;
+    return clipPathSvg;
+  }
+
+  /**
+   * @param {DOM} dom
+   * @param {LH.Artifacts.FullPageScreenshot} fullPageScreenshot
+   */
+  static _installBackgroundImageStyle(dom, fullPageScreenshot) {
+    const containerEl = dom.find('.lh-container', dom._document);
+    if (containerEl.querySelector('#full-page-screenshot-style')) return;
+
+    const fullpageScreenshotUrl = fullPageScreenshot.data;
+    const fullPageScreenshotStyleTag = dom.createElement('style');
+    fullPageScreenshotStyleTag.id = 'full-page-screenshot-style';
+    fullPageScreenshotStyleTag.innerText = `
+      .lh-element-screenshot__image {
+        background-image: url(${fullpageScreenshotUrl})
+      }`;
+    containerEl.appendChild(fullPageScreenshotStyleTag);
+  }
+
+  /**
+   * @param {DOM} dom
+   * @param {ParentNode} templateContext
+   * @param {LH.Artifacts.FullPageScreenshot} fullPageScreenshot
+   */
+  static installOverlayFeature(dom, templateContext, fullPageScreenshot) {
+    ElementScreenshotRenderer._installBackgroundImageStyle(dom, fullPageScreenshot);
+
+    for (const el of dom.document().querySelectorAll('.lh-element-screenshot')) {
+      el.addEventListener('click', () => {
+        const overlay = dom.createElement('div');
+        overlay.classList.add('lh-element-screenshot__overlay');
+        const clipRect = {
+          width: Number(el.getAttribute('rectWidth')),
+          height: Number(el.getAttribute('rectHeight')),
+          left: Number(el.getAttribute('rectLeft')),
+          right: Number(el.getAttribute('rectRight')),
+          top: Number(el.getAttribute('rectTop')),
+          bottom: Number(el.getAttribute('rectBottom')),
+        };
+        overlay.appendChild(ElementScreenshotRenderer.render(
+          dom,
+          templateContext,
+          clipRect,
+          fullPageScreenshot,
+          {
+            // TODO: should this be documentElement width?
+            width: window.innerWidth * 0.75,
```

Comment:
I don't understand these numbers, it's supposed to be `viewportSize` argument but then why is it `0.75` is it supposed to be `maxSizeOfLightboxImage` basically?

Reply:
Gave it a touch up.

It should be the size of the element (on screen) that overlay will be added to. this should work in devtools too but is probably not generic enough to support any arbitrary embedding of the LH report ...